src/library/scala/unchecked.scala

```diff
@@ -15,7 +15,7 @@ package scala
 /** An annotation to designate that the annotated entity
  *  should not be considered for additional compiler checks.
  *  Specific applications include annotating the subject of
- *  a match expression to suppress exhaustiveness warnings, and
+ *  a match expression to suppress exhaustiveness and reachability warnings, and
```

Comment:
Old PR, but this doesn't seem to be the case in Scala 2? You also mention it in https://github.com/scala/scala/pull/9140#issuecomment-699871332.